### PR TITLE
FAQ model and Controller Update

### PR DESCRIPTION
**Pull request by spiray**
I changed the mongo URI to a personal account. I implemented the model and modified the controller to interface with the DB. I am aware that this flow may not be optimal so please provide constructive criticism. Thanks!

### CODE REVIEW

**Comment by spiray** [comment]
[As a Developer, I would like to have the FAQ data in a DB](https://trello.com/c/M6cheHHg/84-as-a-developer-i-would-like-to-have-the-faq-data-in-a-db)

File: src/config/index.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ module.exports = {
   },
   database: {
     uri: process.env.NODE_ENV === 'production'
-      ? process.env.MONGO_URI
-      : 'mongodb://localhost:27017/codinggardencommunity',
+      ? process.env.MONGO_URI : 'mongodb://spiray:abc123abc123@ds139946.mlab.com:39946/codinggardencommunity',
```
Comment (w3cj) [inline src/config/index.js]: Do not include a real mongo address. Anyone can now connect to that DB.

Instead, you can update uri to be this:

```js
uri: process.env.MONGO_URI || 'mongodb://localhost:27017/codinggardencommunity'
```

And then add your mongo connection url in the .env file.

File: src/api/faq/faq.model.js [hunk 2]
```diff
@@ -1 +1,21 @@
...
+const FAQSchema = new Schema({
+  question: {
+    type: String,
+    required: true,
+  },
+  answer: {
+    type: String,
+    required: true,
+  },
+  dateUpdated: {
```
Comment (w3cj) [inline src/api/faq/faq.model.js]: Lets use the standard `createdAt` and `updatedAt` fields. Mongoose can automatically update these.

https://mongoosejs.com/docs/guide.html#timestamps

File: src/api/faq/faq.model.js [hunk 3]
```diff
@@ -1 +1,21 @@
...
+    type: Date,
+    default: Date.now,
+    required: true,
+  },
+});
+
+mongoose.model('faqs', FAQSchema);
```
Comment (w3cj) [inline src/api/faq/faq.model.js]: Update this to:

`module.exports = mongoose.model('faqs', FAQSchema);`

Then in any other file you can directly require the module without having to import mongoose.

File: src/api/faq/faq.controller.js [hunk 4]
```diff
@@ -1,67 +1,32 @@
...
-{
-  id: 7,
-  question: 'How do you add emojis in VS Code?',
-  answer: 'This is a feature of Mac OS X. Press CTRL+CMD+Space to bring up the emoji menu! On Windows 10 you can use CTRL+Period',
-},
-];
+const mongoose = require('mongoose');
 
-function getFAQ(req, res) {
+require('./faq.model');
```
Comment (w3cj) [inline src/api/faq/faq.controller.js]: See comment below about exporting model. After that change, this line can be updated to:

`const FAQ = require('./faq.model');`

And you can remove the mongoose require above.

**Review by w3cj — CHANGES_REQUESTED** [review]
See comments.

File: src/config/index.js (hunk 1 above)
Comment (spiray) [reply src/config/index.js]: Thanks, Done!

File: src/api/faq/faq.model.js (hunk 2 above)
Comment (spiray) [reply src/api/faq/faq.model.js]: Thanks, Done!

File: src/api/faq/faq.model.js (hunk 3 above)
Comment (spiray) [reply src/api/faq/faq.model.js]: Done!

File: src/api/faq/faq.controller.js (hunk 4 above)
Comment (spiray) [reply src/api/faq/faq.controller.js]: Done!

**Comment by spiray** [comment]
@w3cj Thanks for reviewing my PR. I made all of the requested changes. Let me know if you find anything else. Thanks again!

File: src/index.js [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
 require('dotenv').config();
 
+const { contributors, faq } = require('./routes');
+const { server: { port, host } } = require('./config');
+
 // Database connection
 require('./helpers/databaseConnection');
 
+// Seed DB
+require('./tasks/seedFaq');
```
Comment (w3cj) [inline src/index.js]: No need to require this here. Lets add a script to the `package.json` called `seed` that can be run as needed.

**Review by w3cj — CHANGES_REQUESTED** [review]
See comment.

File: src/index.js (hunk 5 above)
Comment (spiray) [reply src/index.js]: Got it now, Done.

Comment (spiray) [reply src/index.js]: See latest commit, is that the flow you had in mind?

**Pull request merged**